data_pump/bitstream.py

```diff
@@ -0,0 +1,166 @@
+            # because the server may be out of memory
+            # fill the tables: most_recent_checksum and checksum_result
+            # based on imported bitstreams
+            try:
+                do_api_post(checksum_url, None, None)
+            except Exception as e:
+                e_json = json.loads(e.args[0])
+                logging.error('POST request ' +
+                              e_json['path'] + ' failed. Status: ' +
+                              str(e_json['status']))
+            counter = 0
+        counter += 1
+        bitstream_json_p = {}
+        metadata_bitstream_dict = \
+            metadata_class.get_metadata_value(0, bitstream['bitstream_id'])
+        if metadata_bitstream_dict is not None:
+            bitstream_json_p['metadata'] = metadata_bitstream_dict
+            # bitstream['size_bytes']
+        bitstream_json_p['sizeBytes'] = 1748
+        # bitstream['checksum']
+        bitstream_json_p['checkSum'] = {
+            'checkSumAlgorithm': bitstream['checksum_algorithm'],
+            'value': '8a4605be74aa9ea9d79846c1fba20a33'
+        }
+        if not bitstream['bitstream_format_id']:
+            logging.info(
+                f'Bitstream {bitstream["bitstream_id"]} '
+                f'does not have a bitstream_format_id. '
+                f'Using {unknown_format_id_val} instead.')
+            bitstream['bitstream_format_id'] = unknown_format_id_val
+            # bitstream['internal_id']
+        params = {'internal_id': '77893754617268908529226218097860272513',
+                  'storeNumber': bitstream['store_number'],
+                  'bitstreamFormat': bitstreamformat_id_dict[
+                      bitstream['bitstream_format_id']],
+                  'deleted': bitstream['deleted'],
+                  'sequenceId': bitstream['sequence_id'],
+                  'bundle_id': None,
+                  'primaryBundle_id': None}
+
+        # if bitstream has bundle, set bundle_id from None to id
+        if bitstream['bitstream_id'] in bitstream2bundle_dict:
+            params['bundle_id'] = \
+                bundle_id_dict[bitstream2bundle_dict[bitstream['bitstream_id']]]
+
+        # if bitstream is primary bitstream of some bundle,
+        # set primaryBundle_id from None to id
+        if bitstream['bitstream_id'] in primary_bitstream_dict:
+            params['primaryBundle_id'] = \
+                bundle_id_dict[primary_bitstream_dict[bitstream['bitstream_id']]]
+        try:
+            logging.info('Going to process Bitstream with internal_id: ' +
+                         str(bitstream['internal_id']))
+            response = do_api_post(bitstream_url, params, bitstream_json_p)
+            bitstream_id_dict[bitstream['bitstream_id']] = \
+                convert_response_to_json(response)['id']
+            imported += 1
+        except Exception:
+            logging.error(
+                'POST request ' + response.url + ' for id: ' +
```

Comment:
response not initialized, use something else. E.g bitstream_url. Now, if it fails during post, no comprehensible message appears and error propagates. 2 more places, exactly the same problem